homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1103,6 +1161,63 @@ async def async_set_dry_off(self):
         )
 
 
+class XiaomiAirHumidifierMiot(XiaomiAirHumidifier):
+    """Representation of a Xiaomi Air Humidifier (MiOT protocol)."""
+
+    @property
+    def speed(self):
+        """Return the current speed."""
+        if self._state:
+            return AirhumidifierMiotOperationMode(self._state_attrs[ATTR_MODE]).name
+
+        return None
+
+    @property
+    def button_pressed(self):
+        """Return the last button pressed."""
+        if self._state:
+            return AirhumidifierPressedButton(
+                self._state_attrs[ATTR_BUTTON_PRESSED]
+            ).name
+
+        return None
+
+    async def async_set_speed(self, speed: str) -> None:
+        """Set the speed of the fan."""
+        if self.supported_features & SUPPORT_SET_SPEED == 0:
```

Comment:
This check is already done by the fan integration. We can remove it.

Reply:
ok
But I didn't see that https://github.com/home-assistant/core/blob/2ef3dfb6738e8dd96b6e89b8b11fa6197de82ef0/homeassistant/components/fan/__init__.py#L115
Are you sure, I can to del that?

Reply:
Yes.